lib/CRUD/sql_screen.dart

```diff
@@ -0,0 +1,165 @@
+    final data = await SQLHelper.getItems();
+    setState(() {
+      _users = data;
+      _isLoading = false;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _refreshusers();
+  }
+
+  final TextEditingController _titleController = TextEditingController();
+  final TextEditingController _descriptionController = TextEditingController();
+
+  
+  void _showForm(int? id) async {
+    if (id != null) {
+      // id == null -> create new item
+      // id != null -> update an existing item
+      final existingUserList =
+      _users.firstWhere((element) => element['id'] == id);
+      _titleController.text = existingUserList['title'];
+      _descriptionController.text = existingUserList['description'];
+    }
+
+    showModalBottomSheet(
+        context: context,
+        elevation: 5,
+        isScrollControlled: true,
+        builder: (_) => Container(
+          padding: EdgeInsets.only(
+            top: 15,
+            left: 15,
+            right: 15,
+            // this will prevent the soft keyboard from covering the text fields
+            bottom: MediaQuery.of(context).viewInsets.bottom + 120,
+          ),
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            crossAxisAlignment: CrossAxisAlignment.end,
+            children: [
+              TextField(
+                controller: _titleController,
+                decoration: const InputDecoration(hintText: 'Title'),
+              ),
+              const SizedBox(
+                height: 10,
+              ),
+              TextField(
+                controller: _descriptionController,
+                decoration: const InputDecoration(hintText: 'Description'),
+              ),
+              const SizedBox(
+                height: 20,
+              ),
+              ElevatedButton(
+                onPressed: () async {
+                  // Save new user
+                  if (id == null) {
+                    await _addItem();
+                  }
+
+                  if (id != null) {
+                    await _updateItem(id);
+                  }
+
+                  // Clear the text fields
+                  _titleController.text = '';
+                  _descriptionController.text = '';
+
+                  // Close the bottom sheet
+                  Navigator.of(context).pop();
+                },
+                child: Text(id == null ? 'Create New' : 'Update'),
+              )
+            ],
+          ),
+        ));
+  }
+
+// Insert a new journal to the database
+  Future<void> _addItem() async {
+    await SQLHelper.createItem(
+        _titleController.text, _descriptionController.text);
+    _refreshusers();
+  }
+
+  // Update an existing journal
+  Future<void> _updateItem(int id) async {
+    await SQLHelper.updateItem(
+        id, _titleController.text, _descriptionController.text);
+    _refreshusers();
+  }
+
+  // Delete an item
+  void _deleteItem(int id) async {
```

Comment:
these functions should be grouped together after build function

Reply:
Done sir